content/blog/host-based-volume-encryption-with-hpe-csi-driver-for-kubernetes.md

```diff
@@ -0,0 +1,249 @@
+  csi.storage.k8s.io/node-publish-secret-name: hpe-backend
+  csi.storage.k8s.io/node-publish-secret-namespace: hpe-storage
+  csi.storage.k8s.io/node-stage-secret-name: hpe-backend
+  csi.storage.k8s.io/node-stage-secret-namespace: hpe-storage
+  csi.storage.k8s.io/provisioner-secret-name: hpe-backend
+  csi.storage.k8s.io/provisioner-secret-namespace: hpe-storage
+  csi.storage.k8s.io/fstype: xfs
+  allowOverrides: hostEncryption,hostEncryptionSecretName,hostEncryptionSecretNamespace
+reclaimPolicy: Delete
+allowVolumeExpansion: true
+```
+
+Now, the user first need to create their own `Secret`.
+
+```yaml
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: my-encryption-key
+  namespace: my-namespace
+stringData:
+  hostEncryptionPassphrase: "This is another secret, it's all mine!"
+```
+
+This `Secret` now needs to be referenced in a set of annotations of the `PersistentVolumeClaim`.
+
+```yaml
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: my-encrypted-pvc
+  annotations:
+    csi.hpe.com/hostEncryption: "true"
+    csi.hpe.com/hostEncryptionSecretName: my-encryption-key
+    csi.hpe.com/hostEncryptionSecretNamespace: my-namespace
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 64Gi
+  storageClassName: hpe-standard-delegated
+```
+
+Let's use the above `StorageClass`, `Secret` and `PersistentVolumeClaim` and attach a workload.
+
+# Provisioning and writing data
+
+In this example I'm just bringing up a sleeping `Pod` and attaching the `PersistentVolumeClaim` named "my-encrypted-pvc" created from the previous section.
+
+```yaml
+---
+kind: Pod
+apiVersion: v1
+metadata:
+  name: my-pod
+spec:
+  containers:
+    - name: my-test-pod
+      image: alpine
+      command: ["/bin/sh"]
+      args: ["-c", "while true; do echo Snoozing...; sleep 10; done"]
+      volumeMounts:
+        - name: my-encrypted-mount
+          mountPath: /data
+  volumes:
+    - name: my-encrypted-mount
+      persistentVolumeClaim:
+        claimName: my-encrypted-pvc
+```
+
+Once the `Pod` has come up, we can inspect the mount point.
+
+```markdown
+$ kubectl exec pod/my-pod -- df -h /data
+Filesystem              Size   Used   Available  Use%  Mounted on
+/dev/mapper/enc-mpatha  64.0G  32.2M  63.9G      0%    /data
+```
+
+The important difference we can observe here versus an unencrypted `PersistentVolumeClaim` is the "enc" keyword in the device name. This means that the kernel module "dm-crypt" has been used to broker access to the actual volume via Linux Unified Key Setup (LUKS). The HPE CSI Driver uses the default cipher with "dm-crypt" which currently is "aes-xts-plain64" with a key size of 256 bits.
+
+# Inspecting the wire
+
+Assume we would write "Hello World" into an unencrypted volume versus an encrypted. In the following experiment I captured `echo "Hello World" > /data/file.txt && sync` with a network packet sniffer and loaded up the result in Wireshark.
```

Comment:
Change the end of the first sentence to read "...volume versus one that is encrypted."

Second sentence, add a comma after "In the following experiment"